src/level/level.cpp

```diff
@@ -238,22 +247,112 @@ namespace level {
         artifacts_.remove(i);
     }
 
+    void Level::enemyShoot(entities::Shooter *s, manager::Level *levelManager){
+        Collidable *collision;
+        bool firable = true;
+        if(s->canShoot()){
```

Comment:
In questo caso forse, la funzione `canShoot` si dovrebbe prendere carico di tutti i check riguardo la linea di tiro, dato che dal nome deve proprio andare a vedere se si può sparare.